src/utils/wallet/bitget_wallet.ts

```
@@ -0,0 +1,174 @@
+    }
+  };
+
+  getWalletProviderName = async (): Promise<string> => {
+    return 'Bitget Wallet';
+  };
+
+  getAddress = async (): Promise<string> => {
+    let accounts = (await this.provider?.getAccounts()) || []
+    return accounts[0]
+  };
+
+  getPublicKeyHex = async (): Promise<string> => {
+    let publicKey = await this.provider?.getPublicKey()
+    if (!publicKey) {
+      console.log('bitget Wallet not connected')
+      return ''
+    }
+    return publicKey
+  };
+
+  signPsbt = async (psbtHex: string): Promise<string> => {
+    const data = {
+      method: 'signPsbt',
+      params: {
+        from: this.provider?.selectedAddress,
+        __internalFunc: '__signPsbt_babylon',
+        psbtHex,
+        options: {
+          autoFinalized: true
+        }
+      }
+    }
+
+    let signedPsbt = await this.provider?.request('dappsSign', data)
+    let psbt = Psbt.fromHex(signedPsbt)
+    try {
+      psbt.finalizeAllInputs()
+    } catch (error) {
+      console.log('babylon-signPsbt failed', error)
+    }
+
+    let transaction = psbt.extractTransaction()
+    return transaction.toHex()
+  };
+
+  signPsbts = async (psbtsHexes: string[]): Promise<string[]> => {
+    if (!psbtsHexes && !Array.isArray(psbtsHexes)) {
+      throw 'parmas error'
+    }
+    const options = psbtsHexes.map(_ => {
+      return {
+        autoFinalized: true
+      }
+    })
+    const data = {
+      method: 'signPsbt',
+      params: {
+        from: this.provider?.selectedAddress,
+        __internalFunc: '__signPsbts_babylon',
+        psbtHex: '_',
+        psbtsHexes,
+        options
+      }
+    }
+
+    try {
+      let signedPsbts = await this.provider?.request('dappsSign', data)
+      signedPsbts = signedPsbts.split(',')
+      return signedPsbts.map((tx:string) => {
+        let psbt = Psbt.fromHex(tx)
+        try {
+          psbt.finalizeAllInputs()
+        } catch (error) {
+          console.log('babylon-signPsbts failed', error)
```

this should throw an error.

---

Same reason with the signPsbt. Should we remove it or leave it as is?

---

Done.